Galaxy Quick failure due to bandwidth result update issue

quick-wh-glx-health fails after the actual tests successfully run:

```
Run tenstorrent/tt-metal/.github/actions/upload-data-via-sftp@main
Run # Change dir to path if its set
Changing dir to /work
Run # Change dir to path if its set
Changing dir to /work
Warning: Permanently added '***' (RSA) to the list of known hosts.
stat generated/fabric/bandwidth_summary_results_*_upload.csv: No such file or directory
sftp> put -r generated/fabric/bandwidth_summary_results_*_upload.csv
Error: Process completed with exit code 1.
```

https://github.com/tenstorrent/tt-metal/actions/runs/20274929250/job/58221674389

--- a/.github/workflows/galaxy-quick-impl.yaml
+++ b/.github/workflows/galaxy-quick-impl.yaml
@@ -94,7 +94,7 @@ jobs:
           if [ -d "generated/fabric" ]; then
             echo "has_benchmark_data=true" >> $GITHUB_OUTPUT
             echo "Benchmark files found, will upload to database"
-          else 
+          else
             echo "has_benchmark_data=false" >> $GITHUB_OUTPUT
             echo "Benchmark files not found, database upload will be skipped"
           fi

--- a/.github/workflows/galaxy-quick-impl.yaml
+++ b/.github/workflows/galaxy-quick-impl.yaml
@@ -86,9 +86,21 @@ jobs:
         if: ${{ !cancelled() }}
         with:
           prefix: "test_reports_"
+      - name: Check for benchmark data
+        id: check-benchmark-data
+        if: ${{ !cancelled() && github.ref_name == 'main' }}
+        shell: bash
+        run: |
+          if [ -d "generated/fabric" ]; then
+            echo "has_benchmark_data=true" >> $GITHUB_OUTPUT
+            echo "Benchmark files found, will upload to database"
+          else 
+            echo "has_benchmark_data=false" >> $GITHUB_OUTPUT
+            echo "Benchmark files not found, database upload will be skipped"
+          fi
       - name: Upload benchmark data
         ## Only upload fabric benchmark data for main branch
-        if: ${{ !cancelled() && github.ref_name == 'main' }}
+        if: ${{ !cancelled() && github.ref_name == 'main' && steps.check-benchmark-data.outputs.has_benchmark_data == 'true' }}
         uses: tenstorrent/tt-metal/.github/actions/upload-data-via-sftp@main
         with:
           ssh-private-key: ${{ secrets.SFTP_BENCHMARK_WRITER_KEY }}

